packages/terra-clinical-header/src/Header.jsx

```diff
@@ -35,40 +35,138 @@ const defaultProps = {
   isSubheader: false,
 };
 
-const Header = ({ title, startContent, endContent, isSubheader, ...customProps }) => {
-  let startElement;
-  if (startContent) {
-    startElement = <div className={cx('flex-end')}>{startContent}</div>;
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleOnResize = this.handleOnResize.bind(this);
+    this.state = {
+      startContentWidth: undefined,
+      endContentWidth: undefined,
+    };
   }
 
-  let titleElement;
-  if (title) {
-    titleElement = (
-      <div className={cx('title-container')}>
-        <h1 className={cx('title')}>
-          {title}
-        </h1>
-      </div>
-    );
+  componentDidMount() {
+    // grab the initial full display widths of the end containers for reference
+    // when resizing the end containers inside handleOnResize
+    if (this.startContainerTarget) {
+      this.startContentInitialWidth = this.startContainerTarget.offsetWidth;
+    }
+    if (this.endContainerTarget) {
+      this.endContentInitialWidth = this.endContainerTarget.offsetWidth;
+    }
+
+    window.addEventListener('resize', this.handleOnResize);
```

Comment:
Should we use the `resize-observer-pollyfill` that we're using in our other responsive components? Listening for the resize event on the window won't work when the header width changes because the parent container resizes. Ex. A slide panel that maximizes and minimizes.